promql/generated_parser.y

```diff
@@ -321,7 +321,7 @@ matrix_selector : expr LEFT_BRACKET duration RIGHT_BRACKET
                         {
                         vs, ok := $1.(*VectorSelector)
                         if !ok{
-                                yylex.(*parser).errorf("matrix selectors only allowed for vector selectors")
+                                yylex.(*parser).errorf("range selectors only allowed for vector selectors")
```

Comment:
ranges only allowed

Reply:
Is the terminology "subquery selector" ok, then?

Reply:
Changed the error message accordingly.

Reply:
There's no such thing as a subquery selector, but I don't see that wording anywhere in this PR.